packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -515,6 +515,22 @@ export interface IRootMergeBlock extends IMergeBlock {
 	mergeTree?: MergeTree;
 }
 
+export function findRootMergeBlock(
+	segmentOrNode: IMergeNode | undefined,
+): IRootMergeBlock | undefined {
+	if (segmentOrNode === undefined) {
+		return undefined;
+	}
+	let maybeRoot: IRootMergeBlock | undefined = segmentOrNode.isLeaf()
+		? segmentOrNode.parent
+		: segmentOrNode;
+	while (maybeRoot?.parent !== undefined) {
+		maybeRoot = maybeRoot.parent;
+	}
+
+	return maybeRoot?.mergeTree !== undefined ? maybeRoot : undefined;
```

Comment:
since this function can return a IRootBlock | undefined, the other function also needs to handle undefined. But it might actually be better to move the assert here, as there should always be a root block

Reply:
its actually a bit complicated, as segmentOrNode can also be undefined, which causes findRootMergeBlock to return undefined. None of this is expected to happen, but we want to be defensive if is does. so i think i'll leave it as is.